x-pack/plugins/endpoint/public/applications/endpoint/index.tsx

```diff
@@ -52,6 +66,101 @@ interface RouterProps {
   store: Store;
   coreStart: CoreStart;
   depsStart: EndpointPluginStartDependencies;
+  ingestManager: IngestManagerSetup;
+}
+
+const IsAppUnavailable: React.FunctionComponent<{
+  ingestManager: IngestManagerSetup;
+  setIsAppUnavailable: React.Dispatch<React.SetStateAction<boolean>>;
+}> = ({ ingestManager, setIsAppUnavailable }) => {
+  (async () => {
+    if (await isIngestManagerInitialized(ingestManager)) {
+      setIsAppUnavailable(false);
+      return null;
+    }
+  })();
+
+  const [hasIngestSetupFinished, setHasIngestSetupFinished] = React.useState(false);
```

Comment:
suggestion that you type all your `useState()` calls. Example:

```suggestion
  const [hasIngestSetupFinished, setHasIngestSetupFinished] = React.useState<boolean>(false);
```